backend/src/main/java/com/example/models/ActiveContract.java

```diff
@@ -0,0 +1,39 @@
+package com.example.models;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.sql.Date;
+
+@Entity
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "active_contracts")
+@IdClass(CompositeKey.class)
+// this should DEFINITELY not work
```

Comment:
🥲

Reply:
for some whatever reason, stackoverflow says that it is working...?

Reply:
but it makes no sense to some extent

Reply:
https://stackoverflow.com/questions/71626588/modeling-entities-in-springboot-while-having-the-database-diagram-using-hibernat

Reply:
https://stackoverflow.com/questions/59695741/jpa-how-to-handle-the-entity-mapping-with-two-foreign-keys-and-those-two-forei